packages/core/logging/core-logging-server-internal/src/appenders/rolling_file/rolling_file_appender.ts

```diff
@@ -36,17 +51,22 @@ export class RollingFileAppender implements DisposableAppender {
   private readonly layout: Layout;
   private readonly context: RollingFileContext;
   private readonly fileManager: RollingFileManager;
-  private readonly policy: TriggeringPolicy;
-  private readonly strategy: RollingStrategy;
+  private readonly triggeringPolicy: TriggeringPolicy;
+  private readonly rollingStrategy: RollingStrategy;
+  private readonly retentionPolicy: RetentionPolicy;
   private readonly buffer: BufferAppender;
 
   constructor(config: RollingFileAppenderConfig) {
     this.context = new RollingFileContext(config.fileName);
     this.context.refreshFileInfo();
     this.fileManager = new RollingFileManager(this.context);
     this.layout = Layouts.create(config.layout);
-    this.policy = createTriggeringPolicy(config.policy, this.context);
-    this.strategy = createRollingStrategy(config.strategy, this.context);
+    this.triggeringPolicy = createTriggeringPolicy(config.policy, this.context);
+    this.rollingStrategy = createRollingStrategy(config.strategy, this.context);
+    this.retentionPolicy = createRetentionPolicy(
+      mergeRetentionPolicyConfig(config.retention, config.strategy),
+      this.context
+    );
```

Comment:
I have to use this `mergeRetentionPolicyConfig` "trick" to support the old syntax and maintain backward compatibility